app/src/main/java/com/cortado/minionify/preference/PreferencesSingleton.java

```diff
@@ -0,0 +1,46 @@
+package com.cortado.minionify.preference;
+
+import android.content.SharedPreferences;
+
+import com.cortado.minionify.mode.AppMode;
+import com.cortado.minionify.mode.exceptions.ModeNotExistsException;
+import com.cortado.minionify.preference.exceptions.PreferenceOperationFailure;
+
+public class PreferencesSingleton {
+    private static PreferencesSingleton INSTANCE = null;
+    private SharedPreferences mPreferences;
```

Comment:
very weird to have non-static fields in the singleton. separate with fitting responsibilities

Reply:
irrelevant